pkg/upgrade/upgrades/permanent_upgrades.go

```diff
@@ -50,39 +49,23 @@ func addRootUser(
 	if err != nil {
 		return err
 	}
-
-	// Upsert the role membership into the table.
-	// We intentionally override any existing entry.
-	return deps.DB.Txn(ctx, addRootToAdminRole)
-}
-
-func addRootToAdminRole(ctx context.Context, txn isql.Txn) error {
 	var upsertStmt string
 	var upsertVals []interface{}
-	{
-		// We query the pg_attribute to determine whether the role_id and member_id
-		// columns are present because we can't rely on version gates here.
```

Comment:
Did you determine that this comment was/is no longer true? I did take a look at #94774 (the PR where it was added) just now and I'm not quite sure what the initial reasoning for it was.

Reply:
I think this comment is not true. Tests failed because they were passing version overrides that are not compatible with the actual schema / initial values used.

They had an active version that is lower than the version gate (due to a version override) but in reality they ran the migration that adds these columns and hence need to pass a non-null value.

In the future, tests passing version overrides should ensure that they also pass an appropriate value for `BootstrapVersionKeyOverride`.